--- gettext-0.12.1/gettext-runtime/intl/relocatable.c	2003-04-07 12:42:24.000000000 +0200
+++ gettext-0.12.1.good/gettext-runtime/intl/relocatable.c	2003-08-26 14:32:46.000000000 +0200
@@ -38,6 +38,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <windef.h>
+#include <wtypes.h>
+#include <winbase.h>
 
 #ifdef NO_XMALLOC
 # define xmalloc malloc
--- gettext-0.16/gettext-tools/gnulib-lib/canonicalize.c	2003-05-09 12:12:06.000000000 +0200
+++ gettext-0.16.good/gettext-tools/gnulib-lib/canonicalize.c	2003-08-26 15:22:31.000000000 +0200
@@ -46,6 +46,7 @@
 #endif
 
 #include <sys/stat.h>
+#define lstat stat
 
 #include <errno.h>
 #ifndef _LIBC
--- gettext-0.16/gettext-tools/gnulib-lib/mkdtemp.c	2003-01-14 14:36:07.000000000 +0100
+++ gettext-0.16.good/gettext-tools/gnulib-lib/mkdtemp.c	2003-08-26 15:08:12.000000000 +0200
@@ -167,7 +167,7 @@
       v /= 62;
       XXXXXX[5] = letters[v % 62];
 
-      fd = __mkdir (tmpl, S_IRUSR | S_IWUSR | S_IXUSR);
+      fd = __mkdir (tmpl);
 
       if (fd >= 0)
 	{
--- gettext-0.16/gettext-tools/gnulib-lib/relocatable.c	2003-04-07 12:42:24.000000000 +0200
+++ gettext-0.16.good/gettext-tools/gnulib-lib/relocatable.c	2003-08-26 14:08:11.000000000 +0200
@@ -38,6 +38,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <windef.h>
+#include <winbase.h>
 
 #ifdef NO_XMALLOC
 # define xmalloc malloc
--- gettext-0.16/gettext-tools/gnulib-lib/xreadlink.c	2003-02-28 13:19:46.000000000 +0100
+++ gettext-0.16.good/gettext-tools/gnulib-lib/xreadlink.c	2003-08-26 15:14:40.000000000 +0200
@@ -66,6 +68,11 @@
    If realloc fails, or if the link value is longer than SIZE_MAX :-),
    give a diagnostic and exit.  */
 
+int readlink(const char *path, char *buf, size_t bufsiz)
+{
+    return -1;
+}
+
 char *
 xreadlink (char const *filename)
 {
--- gettext-0.16/Makefile.am.orig	2006-10-09 13:55:41.000000000 +0200
+++ gettext-0.16/Makefile.am	2006-10-30 07:48:28.900392000 +0100
@@ -19,7 +19,7 @@
 
 AUTOMAKE_OPTIONS = 1.5 gnu no-dependencies
 
-SUBDIRS = gnulib-local gettext-runtime gettext-tools autoconf-lib-link
+SUBDIRS = gnulib-local gettext-runtime autoconf-lib-link
 
 EXTRA_DIST = \
   version.sh PACKAGING ChangeLog.0 autogen.sh \
