--- gettext-0.20.1/configure.ac.orig	2019-06-27 21:32:43.612311259 +0200
+++ gettext-0.20.1/configure.ac	2019-06-27 21:34:51.651617610 +0200
@@ -37,7 +37,7 @@
 
 dnl Checks for library functions.
 
-AC_CONFIG_SUBDIRS([gettext-runtime libtextstyle gettext-tools])
+AC_CONFIG_SUBDIRS([gettext-runtime])
 
 AM_EXTRA_RECURSIVE_TARGETS([maintainer-update-po])
 
--- gettext-0.20.1/Makefile.am.orig	2019-06-27 21:32:43.615644574 +0200
+++ gettext-0.20.1/Makefile.am	2019-06-27 21:35:23.144780330 +0200
@@ -19,7 +19,7 @@
 AUTOMAKE_OPTIONS = 1.5 gnu no-dependencies
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = gnulib-local gettext-runtime libtextstyle gettext-tools
+SUBDIRS = gnulib-local gettext-runtime
 
 changelog_etc = \
   gettext-runtime/ChangeLog.0 \
